walletnode/services/artworkregister/node_test.go

```diff
@@ -0,0 +1,400 @@
+			mockArgs: mockArgs{mock.Anything, mock.AnythingOfType("*artwork.File")},
+			methods:  methods{"UploadImage"},
+			nodes:    Nodes{},
+			nodeAttributes: []*node{
+				{
+					fields:      fields{"3", &mocks.RegisterArtwork{}},
+					methodsCall: methodsCall{1},
+					returnErr:   nil,
+				}, {
+					fields:      fields{"4", &mocks.RegisterArtwork{}},
+					methodsCall: methodsCall{1},
+					returnErr:   fmt.Errorf("failed to open stream"),
+				},
+			},
+			assertion: func(t assert.TestingT, err error, a ...interface{}) bool {
+				//should return first non nil error
+				return assert.Equal(t, "failed to open stream", err.Error())
+			},
+		},
+	}
+
+	for i, testCase := range testCases {
+		testCase := testCase
+
+		t.Run(fmt.Sprintf("testCase-%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			for _, a := range testCase.nodeAttributes {
+				//mock
+				a.fields.registerArtwork.On(testCase.methods.uploadImage,
+					testCase.mockArgs.ctx,
+					testCase.mockArgs.file,
+				).Return(a.returnErr)
+
+				n := &Node{
+					PastelID:        a.fields.pastelID,
+					RegisterArtwork: a.fields.registerArtwork,
+				}
+				testCase.nodes.add(n)
+			}
+
+			testCase.assertion(t, testCase.nodes.sendImage(testCase.args.ctx, testCase.args.file))
+
+			//mock assertion each node
+			for _, a := range testCase.nodeAttributes {
+				a.fields.registerArtwork.AssertExpectations(t)
+				a.fields.registerArtwork.AssertCalled(t, testCase.methods.uploadImage, testCase.args.ctx, testCase.args.file)
+				a.fields.registerArtwork.AssertNumberOfCalls(t, testCase.methods.uploadImage, a.methodsCall.uploadImage)
+			}
+		})
+	}
+}
+
+func TestNodeConnect(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		client          *mocks.Client
+		conn            *mocks.Connection
+		registerArtwork *mocks.RegisterArtwork
+		Address         string
+		PastelID        string
+	}
+
+	type args struct {
+		ctx context.Context
+		err error
+	}
+
+	type methods struct {
+		registerArtwork string
+		connect         string
+	}
+
+	type methodCall struct {
+		connect         int
+		registerArtWork int
+	}
+
+	type mockArgs struct {
+		ctx     interface{}
+		address interface{}
+	}
+
+	testCases := []struct {
+		fields     fields
+		args       args
+		methods    methods
+		methodCall methodCall
+		mockArgs   mockArgs
+		assertion  assert.ErrorAssertionFunc
+	}{
+		{
+			fields: fields{
+				conn:            &mocks.Connection{},
+				client:          &mocks.Client{},
+				registerArtwork: &mocks.RegisterArtwork{},
+				Address:         "127.0.0.1:4444",
+				PastelID:        "1",
+			},
+			methods: methods{
+				registerArtwork: "RegisterArtwork",
+				connect:         "Connect",
+			},
+			methodCall: methodCall{
+				connect:         1,
+				registerArtWork: 1,
+			},
+			mockArgs: mockArgs{mock.Anything, mock.AnythingOfType("string")},
+			args: args{
+				ctx: context.Background(),
+				err: nil,
+			},
+			assertion: assert.NoError,
```

Comment:
You use mock right, it's very good. 
But body of the test func very bloated and poorly readable.
Try to split into a separate functions. e.g. 

`newClient().AcceptAddress("127.0.0.1:4445").SomethingElse()...`
so these `&amp;mocks.Connection{}, &amp;mocks.RegisterArtwork{}` mocks should be inside, we don't need to shove them into cases explicitly, at least in this test.
In order to reuse this mock in other packages you can put into `/walletnode/node/test/client.go`

Reply:
ok got it.